app/Jobs/SendSubscribedNotificationsJob.php

```diff
@@ -31,24 +32,26 @@ public function uniqueId(): string
 
     public function handle(): void
     {
-        $check_ins = $this->phone_number->not_notified_check_ins()->get();
-
-        if ($check_ins->count()) {
-            $this->phone_number->subscriptions()
-                ->each(function (Subscription $subscription) use ($check_ins) {
-                    Mail::to($subscription->user->email)->send(
-                        new CheckInNotificationMail(
-                            check_ins: $check_ins,
-                            phone_number: $this->phone_number
-                        )
-                    );
-
-                    Log::info("Sent notification to {$subscription->user->email}");
-                });
-
-            foreach($check_ins as $check_in) {
-                $check_in->update(['notifications_sent_at' => now()]);
+        Cache::lock('send-notifications:'.$this->phone_number->id, 30)->get(function () {
+            $check_ins = $this->phone_number->not_notified_check_ins()->get();
```

Comment:
I'm not sure we want to send all of the check-ins over email since there's no upper limit. We may just want to let the subscriber know their person has checked in and provide a link to the phone number's show page.

Reply:
I agree, seems sensible to maybe show the latest checkin and link to where the rest are shown.

Reply:
I agree, I think we should just add a link directly to their phone number.

Reply:
Ok. I was thinking that since it can only send from the last 2 minutes, we would be at 120 records if someone sent 1 per second(Which seems unrealistic).

If no one is opposed, I'll put the last 10 that have not been notified, then also put a link to the phone number page too.

Reply:
Just made a new commit here. I'm still uncertain about the "no upper limit" and think this should not be solved here, but on a rate limit for the incoming messages in general. I'm allowing up to 1 per second here per phone number in theory. But, the issue is that we can get someone to spam the system and get 100 per second. I'll take a look at this in a different PR.